### PR TITLE
feat(destination): bqstream - add support for batch of properties

### PR DESCRIPTION
**Pull request by akashrpo**
# Description

> add support to accept array of properties

## Notion Ticket

https://www.notion.so/rudderstacks/bqstream-Add-batching-support-8f9470b6eaac4fc1a2d750bc6827ded0

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Comment by sanpj2292** [comment]
@akashrpo 
Basically, we will need to do two things 
- Move BQStream to routerTransform
- There will always be batching by default
- If not, if there is a single event -> this change will break I think(cause error response). So we will need to tread this way carefully

cc: @utsabc @koladilip

**Review by koladilip — CHANGES_REQUESTED** [review]

File: services/streammanager/bqstream/bqstreammanager.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func (producer *BQStreamProducer) Produce(jsonData json.RawMessage, _ interface{
 	tblId := parsedJSON.Get("tableId").String()
 	props := parsedJSON.Get("properties").String()
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply services/streammanager/bqstream/bqstreammanager.go:90]: Are you saying we will have to send `properties` in the form of an array(from transformer) ?

[the remark this answers is not in the document]
Comment (koladilip) [reply services/streammanager/bqstream/bqstreammanager.go:90]: No, I am suggesting to handle both cases in server only even if you get array or single element
https://pkg.go.dev/github.com/tidwall/gjson#Result.Array - This returns array always. or use:
https://pkg.go.dev/github.com/tidwall/gjson#Result.IsArray and build logic based on it.

[the remark this answers is not in the document]
Comment (sanpj2292) [reply services/streammanager/bqstream/bqstreammanager.go:90]: Got it

[the remark this answers is not in the document]
Comment (akashrpo) [reply services/streammanager/bqstream/bqstreammanager.go]: Built the logic around https://pkg.go.dev/github.com/tidwall/gjson#Result.IsArray

**Review by koladilip — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Comment by koladilip** [comment]
@akashrpo Please add tests for Produce then we can merge.

**Comment by koladilip** [comment]
@akashrpo if all checks are passed you can merge

**Review by koladilip — APPROVED** [review]

**Pull request merged**